pandas/core/tools/datetimes.py

```diff
@@ -229,13 +229,12 @@ def _return_parsed_timezone_results(result, timezones, tz, name):
     -------
     tz_result : Index-like of parsed dates with timezone
     """
-    if tz is not None:
-        raise ValueError(
-            "Cannot pass a tz argument when parsing strings with timezone information."
-        )
     tz_results = np.array(
         [Timestamp(res).tz_localize(zone) for res, zone in zip(result, timezones)]
     )
+    if tz is not None:
```

Comment:
the issue number is not important instead add a comment in what / why u r doing this here